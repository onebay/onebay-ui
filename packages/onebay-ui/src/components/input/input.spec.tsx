import { mount } from '@vue/test-utils'
import Input from './index'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(Input, {
    slots,
    props: { ...values },
  })
}

describe('Input Snap', () => {
  it('render initial Input', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Input -- props style', () => {
    const wrapper = factory({ style: 'border:none;' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Input -- props value', () => {
    const wrapper = factory({ value: 'value' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Input -- props name', () => {
    const wrapper = factory({ name: 'name' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Input -- props placeholder', () => {
    const wrapper = factory({ placeholder: 'placeholder' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Input -- props title', () => {
    const wrapper = factory({ title: 'title' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Input -- props maxLength', () => {
    const wrapper = factory({ maxLength: 10 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Input -- props type(number)', () => {
    const wrapper = factory({ type: 'number' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Input -- props type(password)', () => {
    const wrapper = factory({ type: 'password' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Input -- props type(idcard)', () => {
    const wrapper = factory({ type: 'idcard' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Input -- props type(digit)', () => {
    const wrapper = factory({ type: 'digit' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Input -- props type(phone)', () => {
    const wrapper = factory({ type: 'phone' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Input -- props disabled', () => {
    const wrapper = factory({ disabled: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Input -- props border', () => {
    const wrapper = factory({ border: false })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Input -- props editable', () => {
    const wrapper = factory({ editable: false })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Input -- props error', () => {
    const wrapper = factory({ error: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Input -- props clear', () => {
    const wrapper = factory({ clear: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Input -- props required', () => {
    const wrapper = factory({ required: true, title: 'name' })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('Input Event', () => {
  it('Input onChange', () => {
    const onChange = jest.fn()
    const wrapper = factory({ onChange: onChange })
    wrapper.find('.ob-input__input').trigger('change')
    expect(onChange).toBeCalled()
  })

  it('Input onInput event', () => {
    const wrapper = factory()
    wrapper.find('.ob-input__input').trigger('input')
    expect(wrapper.emitted()).toHaveProperty('input')
  })

  it('Input onFocus', () => {
    const onFocus = jest.fn()
    const wrapper = factory({ onFocus: onFocus })
    wrapper.find('.ob-input__input').trigger('focus')
    expect(wrapper.emitted()).toHaveProperty('focus')
  })

  it('Input onBlur', () => {
    const onBlur = jest.fn()
    const wrapper = factory({ onBlur: onBlur })
    wrapper.find('.ob-input__input').trigger('blur')
    expect(wrapper.emitted()).toHaveProperty('blur')
  })

  it('Input onClick', () => {
    const onClick = jest.fn()
    const wrapper = factory({ onClick: onClick, editable: false })
    wrapper.find('.ob-input .ob-input__overlay').trigger('click')
    expect(onClick).toBeCalled()
  })

  it('Input clear Behavior', () => {
    const onChange = jest.fn()
    const wrapper = factory({ onChange: onChange, clear: true, value: 'value' })
    wrapper.find('.ob-input .ob-input__icon').trigger('click')
    expect(wrapper.vm.inputClearing).toBeTruthy()
    expect(onChange).toBeCalled()
  })

  it('Input error Behavior', () => {
    const onErrorClick = jest.fn()
    const wrapper = factory({ onErrorClick: onErrorClick, error: true, value: 'value' })
    wrapper.find('.ob-input .ob-input__icon').trigger('click')
    expect(onErrorClick).toBeCalled()
  })
})
