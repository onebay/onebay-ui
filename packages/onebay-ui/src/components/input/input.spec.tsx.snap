// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Snap render Input -- props border 1`] = `
<div
  class="ob-input ob-input--without-border"
>
  <div
    class="ob-input__container"
  >
    <div
      class="ob-input__overlay ob-input__overlay--hidden"
    />
    
    <input
      class="ob-input__input"
      id="name"
      name="name"
      placeholder=""
      type="text"
    />
    <!---->
    <!---->
    <div
      class="ob-input__children"
    >
      
      
    </div>
  </div>
</div>
`;

exports[`Input Snap render Input -- props clear 1`] = `
<div
  class="ob-input"
>
  <div
    class="ob-input__container"
  >
    <div
      class="ob-input__overlay ob-input__overlay--hidden"
    />
    
    <input
      class="ob-input__input"
      id="name"
      name="name"
      placeholder=""
      type="text"
    />
    
    <!---->
    <div
      class="ob-input__children"
    >
      
      
    </div>
  </div>
</div>
`;

exports[`Input Snap render Input -- props disabled 1`] = `
<div
  class="ob-input"
>
  <div
    class="ob-input__container ob-input--disabled"
  >
    <div
      class="ob-input__overlay"
    />
    
    <input
      class="ob-input__input"
      id="name"
      name="name"
      placeholder=""
      type="text"
    />
    <!---->
    <!---->
    <div
      class="ob-input__children"
    >
      
      
    </div>
  </div>
</div>
`;

exports[`Input Snap render Input -- props editable 1`] = `
<div
  class="ob-input"
>
  <div
    class="ob-input__container ob-input--disabled"
  >
    <div
      class="ob-input__overlay"
    />
    
    <input
      class="ob-input__input"
      id="name"
      name="name"
      placeholder=""
      type="text"
    />
    <!---->
    <!---->
    <div
      class="ob-input__children"
    >
      
      
    </div>
  </div>
</div>
`;

exports[`Input Snap render Input -- props error 1`] = `
<div
  class="ob-input"
>
  <div
    class="ob-input__container ob-input--error"
  >
    <div
      class="ob-input__overlay ob-input__overlay--hidden"
    />
    
    <input
      class="ob-input__input"
      id="name"
      name="name"
      placeholder=""
      type="text"
    />
    <!---->
    <div
      class="ob-input__icon"
    >
      <div
        class="ob-icon ob-icon-alert-circle ob-input__icon-alert"
      />
    </div>
    <div
      class="ob-input__children"
    >
      
      
    </div>
  </div>
</div>
`;

exports[`Input Snap render Input -- props maxLength 1`] = `
<div
  class="ob-input"
>
  <div
    class="ob-input__container"
  >
    <div
      class="ob-input__overlay ob-input__overlay--hidden"
    />
    
    <input
      class="ob-input__input"
      id="name"
      name="name"
      placeholder=""
      type="text"
    />
    <!---->
    <!---->
    <div
      class="ob-input__children"
    >
      
      
    </div>
  </div>
</div>
`;

exports[`Input Snap render Input -- props name 1`] = `
<div
  class="ob-input"
>
  <div
    class="ob-input__container"
  >
    <div
      class="ob-input__overlay ob-input__overlay--hidden"
    />
    
    <input
      class="ob-input__input"
      id="name"
      name="name"
      placeholder=""
      type="text"
    />
    <!---->
    <!---->
    <div
      class="ob-input__children"
    >
      
      
    </div>
  </div>
</div>
`;

exports[`Input Snap render Input -- props placeholder 1`] = `
<div
  class="ob-input"
>
  <div
    class="ob-input__container"
  >
    <div
      class="ob-input__overlay ob-input__overlay--hidden"
    />
    
    <input
      class="ob-input__input"
      id="name"
      name="name"
      placeholder="placeholder"
      type="text"
    />
    <!---->
    <!---->
    <div
      class="ob-input__children"
    >
      
      
    </div>
  </div>
</div>
`;

exports[`Input Snap render Input -- props required 1`] = `
<div
  class="ob-input"
>
  <div
    class="ob-input__container"
  >
    <div
      class="ob-input__overlay ob-input__overlay--hidden"
    />
    <label
      class="ob-input__title ob-input__title--required"
      for=""
    >
      name
    </label>
    <input
      class="ob-input__input"
      id="name"
      name="name"
      placeholder=""
      type="text"
    />
    <!---->
    <!---->
    <div
      class="ob-input__children"
    >
      
      
    </div>
  </div>
</div>
`;

exports[`Input Snap render Input -- props style 1`] = `
<div
  class="ob-input"
  style=""
>
  <div
    class="ob-input__container"
  >
    <div
      class="ob-input__overlay ob-input__overlay--hidden"
    />
    
    <input
      class="ob-input__input"
      id="name"
      name="name"
      placeholder=""
      type="text"
    />
    <!---->
    <!---->
    <div
      class="ob-input__children"
    >
      
      
    </div>
  </div>
</div>
`;

exports[`Input Snap render Input -- props title 1`] = `
<div
  class="ob-input"
>
  <div
    class="ob-input__container"
  >
    <div
      class="ob-input__overlay ob-input__overlay--hidden"
    />
    <label
      class="ob-input__title"
      for=""
    >
      title
    </label>
    <input
      class="ob-input__input"
      id="name"
      name="name"
      placeholder=""
      type="text"
    />
    <!---->
    <!---->
    <div
      class="ob-input__children"
    >
      
      
    </div>
  </div>
</div>
`;

exports[`Input Snap render Input -- props type(digit) 1`] = `
<div
  class="ob-input"
>
  <div
    class="ob-input__container"
  >
    <div
      class="ob-input__overlay ob-input__overlay--hidden"
    />
    
    <input
      class="ob-input__input"
      id="name"
      name="name"
      placeholder=""
      type="digit"
    />
    <!---->
    <!---->
    <div
      class="ob-input__children"
    >
      
      
    </div>
  </div>
</div>
`;

exports[`Input Snap render Input -- props type(idcard) 1`] = `
<div
  class="ob-input"
>
  <div
    class="ob-input__container"
  >
    <div
      class="ob-input__overlay ob-input__overlay--hidden"
    />
    
    <input
      class="ob-input__input"
      id="name"
      name="name"
      placeholder=""
      type="idcard"
    />
    <!---->
    <!---->
    <div
      class="ob-input__children"
    >
      
      
    </div>
  </div>
</div>
`;

exports[`Input Snap render Input -- props type(number) 1`] = `
<div
  class="ob-input"
>
  <div
    class="ob-input__container"
  >
    <div
      class="ob-input__overlay ob-input__overlay--hidden"
    />
    
    <input
      class="ob-input__input"
      id="name"
      name="name"
      placeholder=""
      type="number"
    />
    <!---->
    <!---->
    <div
      class="ob-input__children"
    >
      
      
    </div>
  </div>
</div>
`;

exports[`Input Snap render Input -- props type(password) 1`] = `
<div
  class="ob-input"
>
  <div
    class="ob-input__container"
  >
    <div
      class="ob-input__overlay ob-input__overlay--hidden"
    />
    
    <input
      class="ob-input__input"
      id="name"
      name="name"
      placeholder=""
      type="text"
    />
    <!---->
    <!---->
    <div
      class="ob-input__children"
    >
      
      
    </div>
  </div>
</div>
`;

exports[`Input Snap render Input -- props type(phone) 1`] = `
<div
  class="ob-input"
>
  <div
    class="ob-input__container"
  >
    <div
      class="ob-input__overlay ob-input__overlay--hidden"
    />
    
    <input
      class="ob-input__input"
      id="name"
      name="name"
      placeholder=""
      type="number"
    />
    <!---->
    <!---->
    <div
      class="ob-input__children"
    >
      
      
    </div>
  </div>
</div>
`;

exports[`Input Snap render Input -- props value 1`] = `
<div
  class="ob-input"
>
  <div
    class="ob-input__container"
  >
    <div
      class="ob-input__overlay ob-input__overlay--hidden"
    />
    
    <input
      class="ob-input__input"
      id="name"
      name="name"
      placeholder=""
      type="text"
    />
    <!---->
    <!---->
    <div
      class="ob-input__children"
    >
      
      
    </div>
  </div>
</div>
`;

exports[`Input Snap render initial Input 1`] = `
<div
  class="ob-input"
>
  <div
    class="ob-input__container"
  >
    <div
      class="ob-input__overlay ob-input__overlay--hidden"
    />
    
    <input
      class="ob-input__input"
      id="name"
      name="name"
      placeholder=""
      type="text"
    />
    <!---->
    <!---->
    <div
      class="ob-input__children"
    >
      
      
    </div>
  </div>
</div>
`;
